types/array_basic.go

```diff
@@ -25,31 +25,18 @@ func newBasicArraySSZ() *basicArraySSZ {
 	}
 }
 
-func (b *basicArraySSZ) Root(val reflect.Value, typ reflect.Type, maxCapacity uint64) ([32]byte, error) {
+func (b *basicArraySSZ) Root(val reflect.Value, typ reflect.Type, fieldName string, maxCapacity uint64) ([32]byte, error) {
```

Comment:
why do you need fieldName ? it isnt used in `basicArraySSZ`

Reply:
I had to add it everywhere because all of these functions need to meet the same interface, and it could be useful down the line for further caching or namespacing